programming.js

```diff
@@ -344,6 +344,14 @@ Numbas.addExtension('programming', ['display', 'util', 'jme'], function(programm
             return this.jobs[job_id];
         }
 
+        /** Get a promise that resolves once all jobs have fully resolved.
+         * @returns {Promise}
+         */
+        get_queue() {
+            const job_promises = Object.entries(this.jobs).map(([k,job]) => job.promise);
```

Comment:
Rather than doing all that fiddling with `Object.entries`, you can just do `Object.values(this.jobs).map(job => job.promise)`